tasks/task_15/scanner/directory_scanner.go

```diff
@@ -0,0 +1,40 @@
+package scanner
+
+import (
+	"context"
+	"fmt"
+	"os"
+)
+
+type DirectoryScanner struct {
+}
+
+func NewDirectoryScanner() *DirectoryScanner {
+	return &DirectoryScanner{}
+}
+
+func (ds *DirectoryScanner) ScanRecursively(ctx context.Context, path string) ([]os.DirEntry, error) {
+	files, err := os.ReadDir(path)
+	if err != nil {
+		return nil, fmt.Errorf("error while reading directory: %w", err)
+	}
+
+	var allFiles []os.DirEntry
+	for _, file := range files {
+		if file.IsDir() {
+			path = path + "/"
```

Comment:
please use correct symbol, event function. `/` won't work on Windows, as there we have `\`